### PR TITLE
Update `onNonMainThreadDetected()` warning

### PR DESCRIPTION
**Pull request by Friendseeker**
Updates the warning to reflect the latest understanding of https://github.com/typelevel/cats-effect/issues/2251 from https://discord.com/channels/632277896739946517/632277897448652844/1310909184937824287

@djspiewak

### CODE REVIEW

**Review by djspiewak — APPROVED** [review]
Is there any way we can detect the sbt version? We're in its JVM in this scenario, so in theory we might be able to putz around with its internals if need be, or at least read envars/properties.

File: core/jvm/src/main/scala/cats/effect/IOApp.scala [hunk 1]
```diff
@@ -350,7 +350,11 @@ trait IOApp {
           """|[WARNING] IOApp `main` is running on a thread other than the main thread.
              |This may prevent correct resource cleanup after `main` completes.
              |This condition could be caused by executing `run` in an interactive sbt session with `fork := false`.
-             |Set `Compile / run / fork := true` in this project to resolve this.
+             |To ensure proper cleanup, either
+             |  - set `Compile / run / fork := true` in this project
+             |  - use `fgRun` instead of `run`
+             |  - use 'bgStop` to terminate `run`
+             |  - update sbt to a version after 1.10.5
```
Comment (armanbilge) [inline core/jvm/src/main/scala/cats/effect/IOApp.scala:357]: Thanks for the PR + in-depth investigation! Sorry if I missed it, what exactly is the change that's landing in sbt > 1.10.5 that fixes this?

Comment (Friendseeker) [reply core/jvm/src/main/scala/cats/effect/IOApp.scala:357]: I also feel grateful for the support I received from you and Daniel during the investigation.

https://github.com/sbt/sbt/pull/7916 is intended to fix it.

The PR is not merged yet, but I did test locally that it makes ctrl-c interrupts properly and the review process seems to be smooth sailing so far.

**Comment by Friendseeker** [comment]
> Is there any way we can detect the sbt version? We're in its JVM in this scenario, so in theory we might be able to putz around with its internals if need be, or at least read envars/properties.

I gave it a look and while I believe it is possible, it is not entirely obvious how to implement it.

sbt does not seem to store sbt version in system properties (I tried `sbt.version`, it has nothing in it). Also sbt seems to have no global object storing version, so reflection to access some global object is no go.

I guess we can search for `build.properties` but at this point it involves File IO, which is also undesirable.

**Pull request merged**